src/string/multiple_longest_common_subsequence.rs

```diff
@@ -0,0 +1,492 @@
+fn matrices_score(chains: &[Vec<char>]) -> Vec<Vec<Vec<u64>>> {
+    let mut scores: Vec<Vec<Vec<u64>>> = vec![];
+    for s1 in chains.iter() {
+        for s2 in chains.iter() {
+            scores.push(score_matrix(s1, s2));
+        }
+    }
+
+    scores
+}
+
+/// Builds the lookup table used for accessing the index of the next char
+/// updates the alphabet to be the alphabet of the letters common to all strings
+///
+/// # Arguments
+/// # 'chains' the strings as a matrix of char
+/// # 'alphabet' the letters in the strings
+///
+/// # Returns
+/// An array of matrices.
+/// Each matrix is tied to a string and can indicate, given a letter,
+/// the next position of that letter in the string.
+fn mt_table(chains: &Vec<Vec<char>>, alphabet: &mut Vec<char>) -> Vec<Vec<Vec<usize>>> {
+    let mut mt: Vec<Vec<Vec<usize>>> = vec![];
+
+    for ch in alphabet.clone() {
+        let mut chain: Vec<Vec<usize>> = vec![];
+
+        for s in chains {
+            let mut v: Vec<usize> = vec![IMPOSSIBLE_NB; s.len()];
+            let mut lpos = IMPOSSIBLE_NB;
+
+            // iterating backwards on the string
+            for i in (0..(s.len())).rev() {
+                if s[i] == ch {
+                    lpos = i;
+                }
+                // pushing the index of the last encounter with the current letter
+                v[i] = lpos;
+            }
+
+            chain.push(v);
+
+            // if the letter was never seen in the current string
+            // then it can't part of the common alphabet
+            if lpos == IMPOSSIBLE_NB {
+                // removing that letter
+                alphabet.retain(|&x| x != ch);
+                chain = vec![];
+                break;
+            }
+        }
+
+        // the letter was seen at leat once
+        if !chain.is_empty() {
+            // pushing an array or array
+            // example on ["AB", "ABAA"]
+            // string1 => {
+            //              'A' => {0, IMPOSSIBLE_NB}
+            //              'B' => {1, 1}
+            //             }
+            // string2 => {
+            //              'A' => {0, 2, 2, 3}
+            //              'B' => {1, 1, IMPOSSIBLE_NB, IMPOSSIBLE_NB}
+            //             }
+            mt.push(chain);
+        }
+    }
+
+    mt
+}
+
+/// Finds one of the longest_common_subsequence among multiple strings
+/// using a similar approach to the A* algorithm in graph theory
+/// [Documentation](https://github.com/epita-rs/MLCS/blob/main/paper.pdf)
+/// # Arguments
+///
+/// * `S` - Array of strings.
+///
+/// # Returns
+///
+/// * `String` if a Longest Common Subsequence exists
+/// * `String' if no LCS was found
+pub fn multiple_longest_common_subsequence(chains: &Vec<&str>) -> String {
+    const C: u64 = 20;
+
+    // Preprocessing
+    let mut ctx = Context::new(chains);
+
+    // queue
+    let mut queue: Vec<Vec<usize>> = vec![];
+    init_queue(&mut ctx, &mut queue);
```

Comment:
This should look more like this:
```suggestion
    let mut queue: Vec<Vec<usize>> = init_queue(&mut ctx);
```